library/alloc/src/slice.rs

```diff
@@ -861,34 +861,37 @@ impl<T: Clone> ToOwned for [T] {
 // Sorting
 ////////////////////////////////////////////////////////////////////////////////
 
-/// Inserts `v[0]` into pre-sorted sequence `v[1..]` so that whole `v[..]` becomes sorted.
+/// Inserts `v[v.len() - 1]` into pre-sorted sequence `v[..v.len() - 1]` so that whole `v[..]` becomes sorted.
 ///
 /// This is the integral subroutine of insertion sort.
 #[cfg(not(no_global_oom_handling))]
-fn insert_head<T, F>(v: &mut [T], is_less: &mut F)
+// benchmarking indicated that inlining makes a substantial improvement, yet only requires a couple of hundred bytes
+#[inline(always)]
```

Comment:
Why not just `#[inline]`?

Reply:
I don't think we can guarantee the performance improvements with just `#[inline]`, as the compiler may decide not to inline it.